### DISCUSSION TITLE
Detect language in the first 60 seconds instead of 30 seconds

### DISCUSSION THREAD

**Discussion by toanhn** [SYSTRAN/faster-whisper#1040] [Q&A]
Hi everyone,
I use the function "transcribe" and it detects language in the first 30 seconds of the audio file. How can I force it to detect language in the first 60 seconds? 
Thanks.

**Comment by Hikari-Tsai** [comment]
Current faster-whisper exposes this directly through `language_detection_segments`. One Whisper window is 30 seconds, so use two segments to inspect up to the first 60 seconds:

```python
segments, info = model.transcribe(
    audio,
    language=None,
    language_detection_segments=2,
)
```

Or call the detection API explicitly:

```python
language, probability, all_probs = model.detect_language(
    audio=audio,
    vad_filter=True,
    language_detection_segments=2,
)
```

With VAD enabled for `detect_language`, silence is removed before the windows are selected, which is usually preferable to simply inspecting 60 wall-clock seconds containing a long silent or musical introduction.

This produces one language choice for the input, not a timeline of languages. For code-switching during transcription, `multilingual=True` is a separate option that performs detection on each transcription window.
